chingu-voyages/v52-tier3-team-34
# Popup event info

This PR introduces the functionality to display popups on map marker clicks and enhances their styling for a better user experience.

**Key features**
- Popups are dynamically rendered when a marker is clicked, showing basic event details: title, description, start and end date.
- Includes a ClickAwayListener to close popups when clicking outside.
- Styled to the popups using TailwindCSS.

**Changes Made**

- Popup functionality:
1. Dynamically display event details in a popup when a marker is clicked.
2. Ensure only one popup is visible at a time.
- Installed the react-click-away-listener library to enable closing popups by clicking outside of it.
- Applied some basic TailwindCSS styling to popups.

**Challenges and Resolutions**
- Unresolved React-Maplibre Popup Bug: Initially, the popups were not rendering correctly due to an issue with the react-maplibre library. This was finally resolved by explicitly setting the closeOnClick property of the Popup component to false.
- Default Behavior Disrupted: Since setting closeOnClick to false disabled the default behaviour of closing popups on outside clicks, I integrated the react-click-away-listener library to restore this functionality.

**Testing Instructions**
- Open the map and click on any marker: Verify that the popup opens, displaying event details (title, description, start, and end times).
- Click outside the popup: Ensure the popup closes correctly using the click-away functionality.
- Click on another marker: Verify that the previously open popup closes and the new one opens.
- Please check for console errors to make sure everything works as expected.
